NeoMathEngine/src/CPU/CpuMathEngineDnnDropout.cpp

```diff
@@ -17,19 +17,51 @@ limitations under the License.
 #pragma hdrstop
 
 #include <MathEngineDnnDropout.h>
+#include <MemoryHandleInternal.h>
 #include <CpuMathEngine.h>
+#include <CpuMathEnginePrivate.h>
 #include <CpuExecutionScope.h>
 #include <CpuRandom.h>
 
 namespace NeoML {
 
-void CCpuMathEngine::Dropout( const CDropoutDesc& dropoutDesc, const CFloatHandle& inputData, const CFloatHandle& outputData )
+CDropoutDesc* CCpuMathEngine::InitDropout(float rate, bool isSpatial, bool isBatchwise)
+{
+	ASSERT_EXPR(rate >= 0.f && rate < 1.f);
+	auto seedDesc = new CSeedDropoutDesc(mathEngine(), true);
```

Comment:
Maybe felds initialization can be moved to ctor to avoid code coping for CPU and CUDA